scripts/bin/revbump

```diff
@@ -1,95 +1,99 @@
-	echo "Usage: revbump [reason for rebuild] [package names] ..."
-	echo
-	echo "Add or increment TERMUX_PKG_REVISION of package."
-	echo
-	exit 1
-fi
-
-REPO_ROOT=$(realpath "$(dirname "$0")/../../")
-IS_GIT_REPOSITORY=false
-
-if git status >/dev/null 2>&1; then
-	IS_GIT_REPOSITORY=true
-fi
-
-for package in "${@:2}"; do
-	package="${package%%/}"
-	buildsh_path=
-	repo=
-	for repo_path in $(jq --raw-output 'keys | .[]' ${REPO_ROOT}/repo.json); do
-		_buildsh_path="${REPO_ROOT}/${repo_path}/${package}/build.sh"
-		repo=$(jq --raw-output ".[\"${repo_path}\"].repo" ${REPO_ROOT}/repo.json)
-		echo $_buildsh_path
-
-		if [ -f "${_buildsh_path}" ]; then
-			buildsh_path=$_buildsh_path
-			break
-		fi
-	done
-
-	if [ -z "$buildsh_path" ]; then
-		echo "${package}: skipping as no build.sh found"
-		continue
-	fi
-
-	if grep -qP '^TERMUX_PKG_REVISION=(\d+)$' "${buildsh_path}"; then
-		# Increment revision.
-		current_rev=$(. "${buildsh_path}" 2>/dev/null; echo "${TERMUX_PKG_REVISION}")
-		incremented_rev=$((current_rev + 1))
-
-		# Replace revision value in build.sh.
-		if ! ${IS_GIT_REPOSITORY}; then
-			echo "${package}: bumping to ${incremented_rev}"
-		fi
-		sed -i -E "s/^(TERMUX_PKG_REVISION=)(.*)$/\1${incremented_rev}/g" "${buildsh_path}"
-	else
-		# Add base (1) revision right after TERMUX_PKG_VERSION.
-		if ! ${IS_GIT_REPOSITORY}; then
-			echo "${package}: setting revision to 1"
-		fi
-		sed -i -E "/TERMUX_PKG_VERSION=(.*)/a TERMUX_PKG_REVISION=1" "${buildsh_path}"
-	fi
-
-	# If we are on Git repository, prompt for committing changes.
-	if ${IS_GIT_REPOSITORY}; then
-		echo
-		echo "You are about to commit these changes:"
-		echo
-		echo "--------------------"
-		git --no-pager diff --patch "${buildsh_path}"
-		echo "--------------------"
-		echo
-		echo "rebuild(${repo}/${package}): $1"
-		echo
-		read -re -p "Do you want to commit these changes ? (y/n) " CHOICE
-		echo
-		if [[ ${CHOICE} =~ (Y|y) ]]; then
-			git add "${buildsh_path}"
-			git commit -m "rebuild(${repo}/${package}): $1"
-		else
-			echo "Not committing to Git!"
-		fi
-		echo
-	fi
-done
+#!/usr/bin/env python3
+import os
+import json
+import sys
+
+filename = sys.argv[0].split("/")[-1]
+
+usage = f"""
+Usage:
+
+{filename} <package> ...            
+
+Add or increment TERMUX_PKG_REVISION of package. 
+
+
+{filename} --dependencies  <package>    
+
+Add or increment TERMUX_PKG_REVISION of all the packages that have <package> in TERMUX_PKG_DEPENDS or in TERMUX_PKG_BUILD_DEPENDS
+"""
+
+
+FOLDERS = list(json.load(open("repo.json")).keys())
+
+
+def get_build_dependent_files(folders: list, DEP : str) -> list:
+    """
+    Gets all the packages that depend on some package
+    """
+    build_files = []
+    for d in folders:
+        for folder in os.listdir(d):
+            for line in open(d+"/"+folder+"/build.sh","r").read().split("\n"):
+                if line.startswith("TERMUX_PKG_DEPENDS") or line.startswith("TERMUX_PKG_BUILD_DEPENDS"):
+                    if DEP in line:
```

Comment:
Just checking if `DEP` is in `line` will also be problematic in some cases, we have for example packages named llvm, llvm-libs, sqlite, libsqlite. `llvm` will be "in" both `llvm-libs` and `llvm`, and `sqlite` in both `sqlite` and `libsqlite`. 

Probably line needs to be split on space, all commas removed, converted into a list, and then we can check if a package is in the list

Reply:
Ok for this I made a function
```python
# here line is dep line
def is_dep(DEP :str,line :str) -> bool:
    """                                                        Checks is the dep is in line
    """                                                        print(line)                                                tmp_deps = line.split("=")[1]
    if tmp_deps.startswith('"'):                                   deps = tmp_deps.split('"')[1].split(", ")              if tmp_deps.startswith("'"):                                   deps = tmp_deps.split("'")[1].split(", ")
    else:                                                          deps = tmp_deps
    return DEP in deps
```